.github/workflows/pr-only-ci.yml

```diff
@@ -0,0 +1,51 @@
+on:
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      pr_number:
+        description: 'Pull Request Number'
+        required: false
+        default: ''
+
+name: PR-only CI
```

Comment:
I added this as a separate file since I don't think this job should run as a merge queue check or on `master` commits (since it's just a warning).

I think other jobs from `ci.yml` could be moved in here, but I didn't do that in this PR.

If this is an annoying distinction to think about, I can just add the job into `ci.yml`.